### PR TITLE
Loading screen

### PR DESCRIPTION
**Pull request by justagoodguy-omega**

### CODE REVIEW

File: app/src/main/java/com/github/onedirection/navigation/fragment/calendar/CustomCalendarView.java [hunk 1]
```diff
@@ -41,13 +42,13 @@
     private final SimpleDateFormat monthFormat = new SimpleDateFormat(("MMM"), Locale.ENGLISH);
     private final List<Date> dates = new ArrayList<>();
     private List<Event> eventsList = new ArrayList<>();
+    private final Activity parentActivity = (Activity) this.getContext();
```
Comment (Ef55) [inline app/src/main/java/com/github/onedirection/navigation/fragment/calendar/CustomCalendarView.java:45]: This doesn't look like a good idea.

File: app/src/main/java/com/github/onedirection/navigation/fragment/calendar/CustomCalendarView.java [hunk 2]
```diff
@@ -166,16 +164,28 @@ private int dateOfMonthAtPosition(int positionInGrid) {
     }
 
     private void callEventCreator(int year, int month, int day) {
-        Intent intent = new Intent(parentFragment.getActivity(), EventCreator.class);
+        Intent intent = new Intent(parentActivity, EventCreator.class);
```
Comment (Ef55) [inline app/src/main/java/com/github/onedirection/navigation/fragment/calendar/CustomCalendarView.java:167]: You could simply use a [context](https://developer.android.com/reference/android/content/Intent?hl=en#Intent(android.content.Context,%20java.lang.Class%3C?%3E)), which would avoid the ugly cast mentioned before.

File: app/src/main/java/com/github/onedirection/navigation/fragment/calendar/CustomCalendarView.java [hunk 3]
```diff
@@ -166,16 +164,28 @@ private int dateOfMonthAtPosition(int positionInGrid) {
...
         EventCreator.putDateExtra(intent, LocalDate.of(year, month, day));
-        parentFragment.startActivity(intent);
+        parentActivity.startActivity(intent);
```
Comment (Ef55) [inline app/src/main/java/com/github/onedirection/navigation/fragment/calendar/CustomCalendarView.java:169]: Same here, the context should be enough.

**Review by Ef55 — CHANGES_REQUESTED** [review]
Happy to see that you learn from your past mistakes e13babf :)

**Review by Sosol02 — APPROVED** [review]
Looks okay.

File: app/src/main/java/com/github/onedirection/navigation/fragment/calendar/CustomCalendarView.java [hunk 4]
```diff
@@ -166,16 +163,28 @@ private int dateOfMonthAtPosition(int positionInGrid) {
...
         //@TODO when the events list is implemented;
     }
 
-    private int getMonthNumber(Calendar cal){
+    private int getMonthNumber(Calendar cal) {
         return cal.get(Calendar.MONTH) + 1;
     }
+
+    private LoadingDialog startLoadingAnimation() {
+        LoadingDialog loadingDialog = new LoadingDialog((Activity) this.getContext());
```
Comment (Ef55) [inline app/src/main/java/com/github/onedirection/navigation/fragment/calendar/CustomCalendarView.java:180]: Moving it doesn't make it better :)

**Review by Ef55 — CHANGES_REQUESTED** [review]
You also included some firebase-emulator configuration files, which should be removed.

File: app/src/main/java/com/github/onedirection/utils/LoadingDialog.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+    public LoadingDialog(Activity hostActivity) {
+        activity = hostActivity;
+    }
+
+    @SuppressLint("InflateParams")
+    public void startLoadingAnimation() {
+
+        AlertDialog.Builder builder = new AlertDialog.Builder(activity);
+        LayoutInflater inflater = activity.getLayoutInflater();
```
Comment (Ef55) [inline app/src/main/java/com/github/onedirection/utils/LoadingDialog.java:22]: (Part of my previous review; dunno why github didn't include it)
You might use 
```suggestion
        LayoutInflater inflater = (LayoutInflater)context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
```
which would solve the aforementioned casting issue, as you did at [CustomCalendarView.java:111](https://github.com/Sosol02/sdp/blob/loadingScreen/app/src/main/java/com/github/onedirection/navigation/fragment/calendar/CustomCalendarView.java#L111).

File: app/src/main/java/com/github/onedirection/navigation/fragment/calendar/CustomCalendarView.java (hunk 4 above)
Comment (justagoodguy-omega) [reply app/src/main/java/com/github/onedirection/navigation/fragment/calendar/CustomCalendarView.java:180]: Well here it literally asks for an activity :'(

**Review by Ef55 — APPROVED** [review]

**Pull request merged**